wgpu-core/src/device/queue.rs

```diff
@@ -42,9 +41,10 @@ use super::{life::LifetimeTracker, Device};
 
 pub struct Queue {
     raw: Box<dyn hal::DynQueue>,
-    pub(crate) device: Arc<Device>,
-    pub(crate) pending_writes: Mutex<ManuallyDrop<PendingWrites>>,
+    pub(crate) pending_writes: Mutex<PendingWrites>,
     life_tracker: Mutex<LifetimeTracker>,
+    // The device needs to be dropped last (`Device.zero_buffer` might be referenced by the encoder in pending writes).
```

Comment:
suggestion(non-blocking): I think it's actually beneficial to use a doc-comment instead here! Ordinary users won't see it, and we can validate the ref. to `zero_buffer` stays valid in CI (which IIRC uses `--document-private-items`):

suggestion: Misc. phrasing/style tweaks.

Combined:

```suggestion
    /// Needs to be dropped last ([`Device::zero_buffer`] might be referenced by the encoder in `pending_writes`).
```

Reply:
I find it confusing that the doc would show up on the field though. The comment is more of a Queue invariant.

Reply:
🤷🏻‍♂️ I don't feel strongly about the third slash or the doc link.